### ISSUE TITLE
aws_vpn_connection raises errors with local_ipv4_network_cidr and remote_ipv4_network_cidr network values less than 32 bits

### ISSUE THREAD

**Issue by cmlccie** [hashicorp/terraform-provider-aws#16879]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform CLI and Terraform AWS Provider Version

```shell
❯ terraform -v
Terraform v0.14.3
+ provider registry.terraform.io/hashicorp/aws v3.22.0
```

### Affected Resource(s)

* aws_vpn_connection

### Terraform Configuration Files

```hcl
variable "profile" {
  type    = string
  default = "default"
}

variable "region" {
  type    = string
  default = "us-east-1"
}

terraform {
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "3.22.0"
    }
  }
}

provider "aws" {
  profile = var.profile
  region  = var.region
}

resource "aws_vpc" "vpc" {
  cidr_block = "10.0.0.0/16"
}

resource "aws_vpn_gateway" "vpn_gateway" {
  vpc_id = aws_vpc.vpc.id
}

resource "aws_customer_gateway" "customer_gateway" {
  bgp_asn    = 65000
  ip_address = "172.0.0.1"
  type       = "ipsec.1"
}

resource "aws_vpn_connection" "main" {
  vpn_gateway_id           = aws_vpn_gateway.vpn_gateway.id
  customer_gateway_id      = aws_customer_gateway.customer_gateway.id
  type                     = "ipsec.1"
  static_routes_only       = true
  local_ipv4_network_cidr  = "192.168.0.0/24"
  remote_ipv4_network_cidr = "10.0.0.0/16"
}
```

### Expected Behavior

The above configuration file should be valid.

### Actual Behavior

```shell
❯ terraform validate

Error: expected "local_ipv4_network_cidr" to contain a network Value with between 32 and 32 significant bits, got: 24

  on test.tf line 45, in resource "aws_vpn_connection" "main":
  45:   local_ipv4_network_cidr  = "192.168.0.0/24"



Error: expected "remote_ipv4_network_cidr" to contain a network Value with between 32 and 32 significant bits, got: 16

  on test.tf line 46, in resource "aws_vpn_connection" "main":
  46:   remote_ipv4_network_cidr = "10.0.0.0/16"
```

### Steps to Reproduce

1. `terraform init`
2. `terraform validate`

### References

* Per the aws provider resource documentation, both [local_ipv4_network_cidr](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/vpn_connection#local_ipv4_network_cidr) and [remote_ipv4_network_cidr](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/vpn_connection#remote_ipv4_network_cidr) should default to `0.0.0.0/0` (0 is less than 32 bits).
* These parameters are configureable through both the AWS console and the [CLI (LocalIpv4NetworkCidr and RemoteIpv4NetworkCidr)](https://awscli.amazonaws.com/v2/documentation/api/latest/reference/ec2/modify-vpn-tunnel-options.html) and should accept any valid IPv4 CIDR with a network value between 0 and 32.
* I believe the issue is in the [aws/resource_aws_vpn_connection.go validateLocalIpv4NetworkCidr()](https://github.com/hashicorp/terraform-provider-aws/blob/22664288907bfac377160ab6cd288a188fd167d2/aws/resource_aws_vpn_connection.go#L1691) function

  ```golang
  func validateLocalIpv4NetworkCidr() schema.SchemaValidateFunc {
	  return validation.All(
		  validation.IsCIDRNetwork(32, 32),
	  )
  }
  ```
  
  I believe this should be modified to accept CIDR networks in the range of 0 to 32:
  
  ```golang
  func validateLocalIpv4NetworkCidr() schema.SchemaValidateFunc {
	  return validation.All(
		  validation.IsCIDRNetwork(0, 32),
	  )
  }
  ```

**Comment by jsf9k** [comment]
I created PR #17573 for this issue.

**Comment by nssol-zhujiafeng** [comment]
I'm using version terraform 0.14.5 and meeting the same bug

**Comment by reznid01** [comment]
I'm using version terraform 0.15.04 and still hitting the same issue. Is there any workarounds?

**Comment by jsf9k** [comment]
> I'm using version terraform 0.15.04 and still hitting the same issue. Is there any workarounds?

There are no workarounds that I know of, short of `apply`ing without `local_ipv4_network_cidr` and `remote_ipv4_network_cidr` and then going into the AWS console to set those values manually.  This issue is resolved by #17573, but I haven't been able to get any traction with that PR.  :disappointed:  If you want to add a :+1: to the PR description that should help increase the likelihood that it will be looked at.

**Comment by reznid01** [comment]
@jsf9k thank you. We are using the CIDR with /32 and after creation changing it to whatever we need.
